tabby-core/src/services/profiles.service.ts

```diff
@@ -36,6 +39,145 @@ export class ProfilesService {
+        }
+
+        const sortKey = p => `${this.resolveProfileGroupName(p.group ?? '')} / ${p.name}`
+        list.sort((a, b) => sortKey(a).localeCompare(sortKey(b)))
+        list.sort((a, b) => (a.isBuiltin ? 1 : 0) - (b.isBuiltin ? 1 : 0))
+        return clone ? deepClone(list) : list
+    }
+
+    /**
+    * Insert a new Profile in config
+    * arg: saveConfig (default: true) -> invoke after the Profile was updated
+    * arg: genId (default: true) -> generate uuid in before pushing Profile into config
+    */
+    async newProfile (profile: PartialProfile<Profile>, saveConfig = true, genId = true): Promise<void> {
+        if (genId) {
+            profile.id = `${profile.type}:custom:${slugify(profile.name)}:${uuidv4()}`
+        }
+
+        const cProfile = this.config.store.profiles.find(p => p.id === profile.id)
+        if (cProfile) {
+            throw new Error(`Cannot insert new Profile, duplicated Id: ${profile.id}`)
+        }
+
+        this.config.store.profiles.push(profile)
+
+        if (saveConfig) {
+            return this.config.save()
+        }
+    }
+
+    /**
+    * Write a Profile in config
+    * arg: saveConfig (default: true) -> invoke after the Profile was updated
+    */
+    async writeProfile (profile: PartialProfile<Profile>, saveConfig = true): Promise<void> {
+        const cProfile = this.config.store.profiles.find(p => p.id === profile.id)
+        if (cProfile) {
+            if (!profile.group) {
+                delete cProfile.group
+            }
+
+            Object.assign(cProfile, profile)
+
+            if (saveConfig) {
+                return this.config.save()
+            }
+        }
+    }
+
+    /**
+    * Delete a Profile from config
+    * arg: saveConfig (default: true) -> invoke after the Profile was deleted
+    */
+    async deleteProfile (profile: PartialProfile<Profile>, saveConfig = true): Promise<void> {
+        this.providerForProfile(profile)?.deleteProfile(this.getConfigProxyForProfile(profile))
+        this.config.store.profiles = this.config.store.profiles.filter(p => p.id !== profile.id)
+
+        const profileHotkeyName = ProfilesService.getProfileHotkeyName(profile)
+        if (this.config.store.hotkeys.profile.hasOwnProperty(profileHotkeyName)) {
+            const profileHotkeys = deepClone(this.config.store.hotkeys.profile)
+            // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+            delete profileHotkeys[profileHotkeyName]
+            this.config.store.hotkeys.profile = profileHotkeys
+        }
+
+        if (saveConfig) {
+            return this.config.save()
+        }
+    }
+
+    /**
+    * Delete all Profiles from config using option filter
+    * arg: options { group: string } -> options used to filter which profile have to be deleted
+    * arg: saveConfig (default: true) -> invoke after the Profile was deleted
+    */
+    async deleteBulkProfiles (options: { group: string }, saveConfig = true): Promise<void> {
```

Comment:
would be more future proof with a predicate function parameter instead of a filter

Reply:
Resolved in 0128013308bbb307275b559b4ce6984581785ebd:
I'm not sure that I've done it the good way. Hope It's okay.